### ISSUE TITLE
Why the output of models are random.

### ISSUE THREAD

**Issue by sjcfr** [huggingface/pytorch-pretrained-BERT#679]
I tried to get word representations using the full-retrained bert model for several times, whereas the outputs of model are different for a same word in each time. Did I neglect something? Not knowing the reason and asking for help sincerely.

The code is:
`from pytorch_pretrained_bert import BertTokenizer, BertModel, BertForMaskedLM `
`model = BertModel.from_pretrained('bert-base-uncased')`
`x=torch.LongTensor([[6541]])`
`y0=model(x)[0]`
`y1=model(x)[0]`

In theory, y0 should be equal to y1. However, they are different.

Both the length of y0 and y1 is 12, in accordance to the 12 layers of 'bert-base-uncased' model. However, each 12 element of y0 and y1 are different.

**Comment by lcswillems** [comment]
As this is not an expected behaviour, they won't be likely to help you if you don't provide a code for reproducing your issue.

**Comment by sjcfr** [comment]
Thanks a lot for reminding.  The issue is renewed with the code.

**Comment by lcswillems** [comment]
That's true! I can reproduce it also on my computer... Really weird!

**Comment by thomwolf** [comment]
You should use `model.eval()` to desactivate dropout like in the usage examples of the readme.

**Comment by sjcfr** [comment]
It solves the problem, thanks!
